apps/arm/main.cpp

```diff
@@ -180,7 +190,76 @@ static mbed_error_status_t setPIDTuningMode(CANMsg &msg) {
 
 // Configure PID parameters
 static mbed_error_status_t setPIDParameter(CANMsg &msg) {
-    return MBED_SUCCESS; // TODO
+    /** ---------------------------------------------------------
+     * | 24 bits  | 32 bits  | 3 bits   | 1 bit   |  4 bits      |
+     * | unused   | data     | param    | vel/pos |  Actuator ID |
+     *  ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+    */
+    // Prepare bitfield to break down CAN data
+    struct{
+        // ID referring to which Actuator should be changed
+        // Param is Kp, Ki, Kd
+        unsigned int actuatorID : 4, param : 3;
+        // Bool for specifying velocity or position pid
+        bool velocity : 1;
+        // Actual value to be set. Bitfields can't have floats, use union
+        int32_t value : 32;
+    } payload;
+    union {
+        float float_value;
+        int32_t int_value;
+    }int_to_float;
+    int64_t data(0);
+    msg >> data;
+    payload.actuatorID = (data & 15);
```

Comment:
when using bitwise operation, it's better to use binary or hex representation, for example, use 0xF, 0b1111 instead of 15. It makes it easier to understand, as it is difficult to convert digital to binary in mind, especially for larger values (1099511627520)